src/Providers/Http/DTO/Request.php

```diff
@@ -0,0 +1,383 @@
+<?php
+
+declare(strict_types=1);
+
+namespace WordPress\AiClient\Providers\Http\DTO;
+
+use InvalidArgumentException;
+use JsonException;
+use WordPress\AiClient\Common\AbstractDataTransferObject;
+use WordPress\AiClient\Providers\Http\Enums\HttpMethodEnum;
+
+/**
+ * Represents an HTTP request.
+ *
+ * This class encapsulates HTTP request data that can be converted
+ * to PSR-7 requests by the HTTP transporter.
+ *
+ * @since n.e.x.t
+ *
+ * @phpstan-type RequestArrayShape array{
+ *     method: string,
+ *     uri: string,
+ *     headers: array<string, list<string>>,
+ *     data?: string|array<string, mixed>|null
+ * }
+ *
+ * @extends AbstractDataTransferObject<RequestArrayShape>
+ */
+class Request extends AbstractDataTransferObject
+{
+    public const KEY_METHOD = 'method';
+    public const KEY_URI = 'uri';
+    public const KEY_HEADERS = 'headers';
+    public const KEY_DATA = 'data';
+
+    /**
+     * @var HttpMethodEnum The HTTP method.
+     */
+    protected HttpMethodEnum $method;
+
+    /**
+     * @var string The request URI.
+     */
+    protected string $uri;
+
+    /**
+     * @var array<string, list<string>> The request headers.
+     */
+    protected array $headers;
+
+    /**
+     * @var array<string, string> Map of lowercase header names to actual header names for fast lookup.
+     */
+    protected array $headersMap;
+
+    /**
+     * @var string|array<string, mixed>|null The request data.
+     */
+    protected $data;
+
+    /**
+     * Constructor.
+     *
+     * @since n.e.x.t
+     *
+     * @param HttpMethodEnum $method The HTTP method.
+     * @param string $uri The request URI.
+     * @param array<string, string|list<string>> $headers The request headers.
+     * @param string|array<string, mixed>|null $data The request data.
+     *
+     * @throws InvalidArgumentException If the URI is empty.
+     */
+    public function __construct(HttpMethodEnum $method, string $uri, array $headers = [], $data = null)
+    {
+        if (empty($uri)) {
+            throw new InvalidArgumentException('URI cannot be empty.');
+        }
+
+        $this->method = $method;
+        $this->uri = $uri;
+        $this->headers = $this->normalizeHeaders($headers);
+        $this->headersMap = $this->buildHeadersMap($this->headers);
+        $this->data = $data;
```

Comment:
How about this?
```suggestion
        if (is_string($data) {
            $this->body = $data;
        } else {
            $this->data = $data;
        }
```

That keeps our properties type safe. If `$data` is a string, it's always a body ready to send, no processing needed.

Reply:
Resolved in 720c294